corehq/apps/es/REINDEX_PROCESS.md

```diff
@@ -0,0 +1,366 @@
+        ES_GROUPS_INDEX_SWAPPED = True
+        ES_SMS_INDEX_SWAPPED = True
+        ES_USERS_INDEX_SWAPPED = True
+        ```
+    4. From the control machine, run update-config and restart commcare services so that the new settings are applied. This will also restart the pillows that were stopped in the previous step.
+        ```
+        cchq <env> update-config
+        cchq <env> service commcare restart
+        ```
+        After this step, the indexes will be swapped and reads will now happen from the newly created secondary indices.
+
+    5. Look around in CommcareHQ, test out things dealing with ES like Submit a form and ensure that it comes up in reports. If you have metrics setup for pillows, verify that the change feed is looking good.
+
+    6. It is recommended to keep the indices in this state for at least 2 working days. This will provide a safe window to fall back if needed.
+
+6. When you are confident that things are working fine with the new index, you are all set to turn off the multiplexer settings. Update `environments/<env>/public.yml` with following values:
+    ```
+    ES_APPS_INDEX_MULTIPLEXED = False
+    ES_CASE_SEARCH_INDEX_MULTIPLEXED = False
+    ES_CASES_INDEX_MULTIPLEXED = False
+    ES_DOMAINS_INDEX_MULTIPLEXED = False
+    ES_FORMS_INDEX_MULTIPLEXED = False
+    ES_GROUPS_INDEX_MULTIPLEXED = False
+    ES_SMS_INDEX_MULTIPLEXED = False
+    ES_USERS_INDEX_MULTIPLEXED = False
+    ```
+
+    From control machine, run update-config and restart commcare services so that new settings are applied.
+
+    ```
+    cchq <env> update-config
+    cchq <env> service commcare restart
+    ```
+
+7. At this stage, writes have been stopped on primary indices. The older indices are eligible for deletion. It is recommended to wait for 6 hours after the multiplexer is turned off. The older indices can be deleted by running.
+
+    ```
+    cchq <env> django-manage elastic_sync_multiplexed delete <index_cname>
+    ```
+8. Delete any lingering residual indices by running
+
+```
+cchq <env> django-manage elastic_sync_multiplexed remove_residual_indices
+```
+These indices are safe to delete and are not used in any functionality of CommCareHQ.
+
+9. Congratulations :tada: You have successfully created new indexes that are active on CommcareHQ.
+
+10. Update [Index Names](./const.py#L27-L57), set Primary index names to secondary index name and secondary index names to `None`.
+
+11. Set Index swapped variables to False
+
+    ```
+    ES_APPS_INDEX_SWAPPED = False
+    ES_CASE_SEARCH_INDEX_SWAPPED = False
+    ES_CASES_INDEX_SWAPPED = False
+    ES_DOMAINS_INDEX_SWAPPED = False
+    ES_FORMS_INDEX_SWAPPED = False
+    ES_GROUPS_INDEX_SWAPPED = False
+    ES_SMS_INDEX_SWAPPED = False
+    ES_USERS_INDEX_SWAPPED = False
+
+    ```
+
+12. Before deploying the changes in Step 10, run update-config.
+
+```
+cchq <env> update-config
+```
+Do not restart the services, let deploy process handle that for you.
```

Comment:
Is it obvious that the user following along should end these steps with a deploy? This makes it seem like they should avoid restarting services because of an imminent deploy, as do the preceding steps, so maybe it makes sense to have the final step be to deploy?

Reply:
That is good point. I was just thinking about the Dimagi process where we are doing the weekly deploy. But I like your point. I will update the docs accordingly.